consensus/state.go

```diff
@@ -1525,7 +1524,9 @@ func (cs *ConsensusState) addVote(vote *types.Vote, peerID p2p.ID) (added bool,
 		blockID, ok := precommits.TwoThirdsMajority()
 		if ok {
 			if len(blockID.Hash) == 0 {
-				cs.enterNewRound(height, vote.Round+1)
+				cs.enterNewRound(height, vote.Round)
+				cs.enterPrecommit(height, vote.Round)
+				cs.enterPrecommitWait(height, vote.Round)
```

Comment:
This could be a separate PR but I made the change here so... LMK if you want me to fork it into a separate PR.